webextensions/README.md

```diff
@@ -0,0 +1,36 @@
+# browser-compat-data
+
+This folder contains compatibility data for the [WebExtensions](https://developer.mozilla.org/Add-ons/WebExtensions) APIs.
+
+It's a JSON file, structured as follows:
+
+* one top level JSON object, which has one property for each WebExtensions API. The name of the property is the name of the module (for example, `"i18n"`, or `"downloads"`). The value is an `api` object.
+
+* an `api` object has one property for each API element (each method, property, type, and event). The name of the property is the name of the element. For example, `"getMessage"` or `"create"`. The value is an `api-element` object.
+
+* an `api-element` is an object with two properties, named: `"desktop_browsers"` and `"mobile_browsers"`. Each of these is a `browser-support` object.
```

Comment:
Ah.

So yes, this schema expects a simple 2-level structure. I think this is actually OK for something like windows.create.state, because that's not a 3-level structure: it's a function ([windows.create](/en-US/Add-ons/WebExtensions/API/windows/create)) taking a type ([windows.windowState](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/Windows/WindowState)). These things each have their docs page under API/windows. There are lots of examples of that and it seems to work fine.

But it is a problem for [storage.StorageArea](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/Storage/StorageArea), which is an object in the storage API that has its own structure, for [storage.StorageArea.get](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/Storage/StorageArea/get) and so on. 3 levels.

storage.StorageArea is I think the only API like this, out of 300 or so. I looked at it, and decided that while we could support N levels, it would add complexity to the schema and to any code that builds compat tables based on the schema, and that this complexity wasn't worth adding for the sake of 1 API. So I made a hack for this, in which you can specify multiple levels as "item.subitem" (like this: https://github.com/mdn/browser-compat-data/blob/master/webextensions/browser-compat-data.json#L4596) and place them directly under the API. Then the code that uses the JSON can use that as the key: https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/storage#Browser_compatibility.

This isn't perfect, but it's simple and seems to work OK. I think if we had lots of 3-level APIs, we might need to think again (but actually, 2-level APIs is an assumption that's quite baked into the docs, not just the schema).